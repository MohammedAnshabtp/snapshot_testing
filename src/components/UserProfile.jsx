import React from "react";
import PropTypes from'prop-types';

function UserProfile({ name, age, hobbies }) {
  return (
    <div>
      <h1>{name}</h1>
      <p>Age:{age}</p>
      {hobbies.length > 0 ? (
        <ul>
          {hobbies.map((hobby, index) => (
            <li key={index}>{hobby}</li>
          ))}
        </ul>
      ) : (
        <p>No Hobbies</p>
      )}
    </div>
  );
}

UserProfile.propTypes ={
    name:PropTypes.string.isRequired,
    age:PropTypes.number.isRequired,
    hobbies:PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default UserProfile;
