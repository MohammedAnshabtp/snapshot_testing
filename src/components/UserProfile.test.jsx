import {render} from "@testing-library/react";
import UserProfile from "../components/UserProfile";
import {describe,it,expect} from 'vitest';

describe('UserProfile Component',()=>{
    it('render correctly with hobbies',()=>{
        const userProps = {
            name: "John",
            age: 25,
            hobbies: ["Sports", "Cooking"],
        };
        const {asFragment} = render(<UserProfile {...userProps} />)
        expect(asFragment()).toMatchSnapshot();
    });

    it('render correctly without hobbies',() =>{
        const userProps = {
            name: "John",
            age: 25,
            hobbies: [],
        };
        const {asFragment} = render(<UserProfile {...userProps} />)
        expect(asFragment()).toMatchSnapshot();
    })
})
