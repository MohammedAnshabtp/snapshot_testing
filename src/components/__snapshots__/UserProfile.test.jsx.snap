// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserProfile Component > render correctly with hobbies 1`] = `
<DocumentFragment>
  <div>
    <h1>
      John
    </h1>
    <p>
      Age:25
    </p>
    <ul>
      <li>
        Sports
      </li>
      <li>
        Cooking
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`UserProfile Component > render correctly without hobbies 1`] = `
<DocumentFragment>
  <div>
    <h1>
      John
    </h1>
    <p>
      Age:25
    </p>
    <p>
      No Hobbies
    </p>
  </div>
</DocumentFragment>
`;
